### PR TITLE
Ex3_Sprint9-JS

### PR DESCRIPTION
**Pull request by DANIEL886886**

### CODE REVIEW

File: ex3/ex3-index.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+// Output: 2
+// Notă: Nu se ține cont de scrierea cu majuscule sau minuscule.
+// Litera A și litera a trebuie numărate la calcularea numărului de apariții pentru A sau pentru a
+//.........................................................................................................
+
+// const countLetterOccurrences = (string, letter) => (typeof string === "string") && (typeof letter === "string") ? for char in string ? count++ : "Invalid input: all input must be string";
+console.clear()
+
+const countLetterOccurrences = (string, letter) => {
+  if (typeof string === "string" && typeof letter === "string") {
```
Comment (MarcelBuian) [inline ex3/ex3-index.js]: ar fi mult mai frumos sa facem validarea la inceput:

if (typeof string !== "string" || typeof letter === "string") {
  throw "Both arguments must be strings.";
}

si apoi incepem codul fara `if`

File: ex3/ex3-index.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    let count = 0;
+       for (let i = 0; i < string.length; i++) {
```
Comment (MarcelBuian) [inline ex3/ex3-index.js]:
```suggestion
    for (let i = 0; i < string.length; i++) {
```
acest for trebuie sa fie identat in acelasi rand cu `let`

**Review by MarcelBuian — APPROVED** [review]
Good work!

**Comment by DANIEL886886** [comment]
const countLetterOccurrences = (string, letter) => {
  let count = 0;
  letter = letter.toLowerCase()

  if (typeof inputstring !== "string" && typeof inputletter !== "string") {
    throw new Error ("Both arguments must be strings.");
  }

  // string.toLowerCase function to convert all letters in lower case in order to count all letters 
  for (let i = 0; i < string.length; i++) {
    if (string[i].toLowerCase() === letter) {
      count++;
    }
  }
  return count;
};

Am refacut structura cu ce mi-ai spus in comentarii si sper ca am reusit sa ordonez si ca identare

**Review by MarcelBuian — APPROVED** [review]
Great work!

**Comment by MarcelBuian** [comment]
Da, arata bine

**Pull request merged**